node-daemon/RPC_INTERFACE.md

```diff
@@ -0,0 +1,1170 @@
+# RPC documentation for Mintlayer node
+
+Version `0.3.0`.
+
+## Module `node`
+
+RPC methods controlling the node.
+
+
+### Method `node_shutdown`
+
+Order the node to shutdown
+
+
+Parameters:
+```
+{}
+```
+
+Returns:
+```
+nothing
+```
+
+
+### Method `node_version`
+
+Get node software version
+
+
+Parameters:
+```
+{}
+```
+
+Returns:
+```
+string
+```
+
+
+### Method `node_set_mock_time`
+
+Set mock time for the node (for testing purposes only)
+
+
+Parameters:
+```
+{
+    "time": number,
+}
+```
+
+Returns:
+```
+nothing
+```
+
+
+## Module `chainstate`
+
+### Method `chainstate_best_block_id`
+
+Get the best block ID
+
+
+Parameters:
+```
+{}
+```
+
+Returns:
+```
+hex string
+```
+
+
+### Method `chainstate_block_id_at_height`
+
+Get block ID at given height in the mainchain
+
+
+Parameters:
+```
+{
+    "height": number,
+}
+```
+
+Returns:
+```
+hex string OR null
+```
+
+
+### Method `chainstate_get_block`
+
+Returns a hex-encoded serialized block with the given id.
+
+
+Parameters:
+```
+{
+    "id": hex string,
+}
+```
+
+Returns:
+```
+hex string OR null
+```
+
+
+### Method `chainstate_get_block_json`
+
+Returns a json-encoded serialized block with the given id.
+
+
+Parameters:
+```
+{
+    "id": hex string,
+}
+```
+
+Returns:
+```
+json OR null
+```
+
+
+### Method `chainstate_get_mainchain_blocks`
+
+Returns a hex-encoded serialized blocks from the mainchain starting from a given block height.
+
+
+Parameters:
+```
+{
+    "from": number,
+    "max_count": number,
+}
+```
+
+Returns:
+```
+[ hex string, .. ]
+```
+
+
+### Method `chainstate_get_utxo`
+
+Returns the TxOutput for a specified UtxoOutPoint.
+
+
+Parameters:
+```
+{
+    "outpoint": {
+        "id": {
+            "BlockReward": hex string,
+        } OR {
+            "Transaction": hex string,
+        },
+        "index": number,
+    },
+}
+```
+
+Returns:
+```
+object OR null
+```
+
+
+### Method `chainstate_submit_block`
+
+Submit a block to be included in the chain
+
+
+Parameters:
+```
+{
+    "block_hex": hex string,
+}
+```
+
+Returns:
+```
+nothing
+```
+
+
+### Method `chainstate_invalidate_block`
+
+Invalidate the specified block and its descendants.
+
+
+Parameters:
+```
+{
+    "id": hex string,
+}
+```
+
+Returns:
+```
+nothing
+```
+
+
+### Method `chainstate_reset_block_failure_flags`
+
+Reset failure flags for the specified block and its descendants.
+
+
+Parameters:
+```
+{
+    "id": hex string,
+}
+```
+
+Returns:
+```
+nothing
+```
+
+
+### Method `chainstate_block_height_in_main_chain`
+
+Get block height in main chain
+
+
+Parameters:
+```
+{
+    "block_id": hex string,
+}
+```
+
+Returns:
+```
+number OR null
```

Comment:
I think this is a good example of docs that can be improved. We should explain why this can be null. Tomorrow I'll drill through them all.